### ISSUE TITLE
'LayerNorm' object has no attribute 'affine'

### ISSUE THREAD

**Issue by liuzimo** [ultralytics/ultralytics#3593]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

root@autodl-container-ac48118452-1a342248:~/autodl-tmp# python train-rt-detr.py
rt-detr-x summary: 867 layers, 67467852 parameters, 0 gradients
Ultralytics YOLOv8.0.131 🚀 Python-3.8.10 torch-1.13.1+cu116 CUDA:0 (NVIDIA GeForce RTX 3090, 24268MiB)
yolo/engine/trainer: task=detect, mode=train, model=None, data=coco.yaml, epochs=300, patience=10, batch=-1, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=None, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=False, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, vid_stride=1, line_width=None, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, v5loader=False, tracker=botsort.yaml, save_dir=/root/autodl-tmp/ultralytics/runs/detect/train
WARNING ⚠️ no model scale passed. Assuming scale='x'.

                   from  n    params  module                                       arguments
  0                  -1  1     25792  ultralytics.nn.modules.block.HGStem          [3, 32, 64]
  1                  -1  6    259200  ultralytics.nn.modules.block.HGBlock         [64, 64, 128, 3, 6]
  2                  -1  1      1408  ultralytics.nn.modules.conv.DWConv           [128, 128, 3, 2, 1, False]
  3                  -1  6   1248256  ultralytics.nn.modules.block.HGBlock         [128, 128, 512, 3, 6]
  4                  -1  6   1788928  ultralytics.nn.modules.block.HGBlock         [512, 128, 512, 3, 6, False, True]
  5                  -1  1      5632  ultralytics.nn.modules.conv.DWConv           [512, 512, 3, 2, 1, False]
  6                  -1  6   2079232  ultralytics.nn.modules.block.HGBlock         [512, 256, 1024, 5, 6, True, False]
  7                  -1  6   2472448  ultralytics.nn.modules.block.HGBlock         [1024, 256, 1024, 5, 6, True, True]
  8                  -1  6   2472448  ultralytics.nn.modules.block.HGBlock         [1024, 256, 1024, 5, 6, True, True]
  9                  -1  6   2472448  ultralytics.nn.modules.block.HGBlock         [1024, 256, 1024, 5, 6, True, True]
 10                  -1  6   2472448  ultralytics.nn.modules.block.HGBlock         [1024, 256, 1024, 5, 6, True, True]
 11                  -1  1     11264  ultralytics.nn.modules.conv.DWConv           [1024, 1024, 3, 2, 1, False]
 12                  -1  6   8221696  ultralytics.nn.modules.block.HGBlock         [1024, 512, 2048, 5, 6, True, False]
 13                  -1  6   9794560  ultralytics.nn.modules.block.HGBlock         [2048, 512, 2048, 5, 6, True, True]
 14                  -1  1    787200  ultralytics.nn.modules.conv.Conv             [2048, 384, 1, 1, None, 1, 1, False]
 15                  -1  1   2168192  ultralytics.nn.modules.transformer.AIFI      [384, 2048, 8]
 16                  -1  1    148224  ultralytics.nn.modules.conv.Conv             [384, 384, 1, 1]
 17                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 18                  10  1    393984  ultralytics.nn.modules.conv.Conv             [1024, 384, 1, 1, None, 1, 1, False]
 19            [-2, -1]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 20                  -1  3   5019648  ultralytics.nn.modules.block.RepC3           [768, 384, 3]
 21                  -1  1    148224  ultralytics.nn.modules.conv.Conv             [384, 384, 1, 1]
 22                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 23                   4  1    197376  ultralytics.nn.modules.conv.Conv             [512, 384, 1, 1, None, 1, 1, False]
 24            [-2, -1]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 25                  -1  3   5019648  ultralytics.nn.modules.block.RepC3           [768, 384, 3]
 26                  -1  1   1327872  ultralytics.nn.modules.conv.Conv             [384, 384, 3, 2]
 27            [-1, 21]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 28                  -1  3   5019648  ultralytics.nn.modules.block.RepC3           [768, 384, 3]
 29                  -1  1   1327872  ultralytics.nn.modules.conv.Conv             [384, 384, 3, 2]
 30            [-1, 16]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 31                  -1  3   5019648  ultralytics.nn.modules.block.RepC3           [768, 384, 3]
 32        [25, 28, 31]  1   7564556  ultralytics.nn.modules.head.RTDETRDecoder    [80, [384, 384, 384]]
rt-detr-x summary: 867 layers, 67467852 parameters, 67467852 gradients

Transferred 1241/1241 items from pretrained weights
TensorBoard: Start with 'tensorboard --logdir /root/autodl-tmp/ultralytics/runs/detect/train', view at http://localhost:6006/
AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
Downloading https://github.com/ultralytics/assets/releases/download/v0.0.0/yolov8n.pt to yolov8n.pt...
⚠️ Download failure, retrying 1/3 https://github.com/ultralytics/assets/releases/download/v0.0.0/yolov8n.pt...
#################################################################################################################################################### 100.0%AMP: checks passed ✅
AutoBatch: Computing optimal batch size for imgsz=640
AutoBatch: CUDA:0 (NVIDIA GeForce RTX 3090) 23.70G total, 0.53G reserved, 0.52G allocated, 22.65G free
      Params      GFLOPs  GPU_mem (GB)  forward (ms) backward (ms)                   input                  output
'LayerNorm' object has no attribute 'affine'
'LayerNorm' object has no attribute 'affine'
'LayerNorm' object has no attribute 'affine'
'LayerNorm' object has no attribute 'affine'
'LayerNorm' object has no attribute 'affine'
AutoBatch: WARNING ⚠️ error detected: expected non-empty vector for x,  using default batch-size 16.
train: Scanning /root/autodl-tmp/datasets/coco/labels/train2017... 117266 images, 1021 backgrounds, 0 corrupt: 100%|██████████| 118287/118287 [01:27<00:00,train: New cache created: /root/autodl-tmp/datasets/coco/labels/train2017.cache
val: Scanning /root/autodl-tmp/datasets/coco/labels/val2017... 4952 images, 48 backgrounds, 0 corrupt: 100%|██████████| 5000/5000 [00:06<00:00, 750.47it/s]val: New cache created: /root/autodl-tmp/datasets/coco/labels/val2017.cache
Plotting labels to /root/autodl-tmp/ultralytics/runs/detect/train/labels.jpg...
optimizer: SGD(lr=0.01, momentum=0.9) with parameter groups 193 weight(decay=0.0), 256 weight(decay=0.0005), 276 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 8 dataloader workers
Logging results to /root/autodl-tmp/ultralytics/runs/detect/train
Starting training for 300 epochs...

      Epoch    GPU_mem  giou_loss   cls_loss    l1_loss  Instances       Size
  0%|          | 0/7393 [00:00<?, ?it/s]/root/miniconda3/lib/python3.8/site-packages/torch/nn/functional.py:1967: UserWarning: nn.functional.sigmoid is deprecated. Use torch.sigmoid instead.
  warnings.warn("nn.functional.sigmoid is deprecated. Use torch.sigmoid instead.")
      1/300      18.8G     0.5135     0.7466     0.2016        145        640:   2%|▏         | 123/7393 [01:21<1:16:23,  1.59it/s]

### Environment

Ultralytics YOLOv8.0.131 🚀 Python-3.8.10 torch-1.13.1+cu116 CUDA:0 (NVIDIA GeForce RTX 3090, 24268MiB)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by liuzimo** [comment]
![image](https://github.com/ultralytics/ultralytics/assets/25811500/359b1ad4-4e4a-474f-81d9-a9647785f7c9)

**Comment by glenn-jocher** [comment]
@liuzimo 👋 hi, thanks for letting us know about this possible problem with YOLO 🚀. We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub with `git pull` or `pip install -U ultralytics` to ensure your problem has not already been solved in the latest code version.
- ✅ **Unmodified** – Your problem must be reproducible using official Ultrlaytics code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/ultralytics/issues/new/choose) with a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃
